server/norace_test.go

```diff
@@ -9599,3 +9599,58 @@ func TestNoRaceJetStreamKVReplaceWithServerRestart(t *testing.T) {
 		t.Fatalf("Encountered errors")
 	}
 }
+
+func TestNoRaceMemStoreCompactDuration(t *testing.T) {
+	//Load MemStore so that it is full
+	subj, msg := "foo", make([]byte, 1000)
+	storedMsgSize := memStoreMsgSize(subj, nil, msg)
+
+	toStore := uint64(500_000)
+	maxBytes := storedMsgSize * toStore
+
+	ms, err := newMemStore(&StreamConfig{Storage: MemoryStorage, MaxBytes: int64(maxBytes)})
+	require_NoError(t, err)
+	defer ms.Stop()
+
+	for i := uint64(0); i < toStore; i++ {
+		ms.StoreMsg(subj, nil, msg)
+	}
+	state := ms.State()
+	if state.Msgs != toStore {
+		t.Fatalf("Expected %d msgs, got %d", toStore, state.Msgs)
+	}
+	if state.Bytes != storedMsgSize*toStore {
+		t.Fatalf("Expected bytes to be %d, got %d", storedMsgSize*toStore, state.Bytes)
+	}
+
+	toStoreOnTop := uint64(10_000_000)
+
+	//1st run: Load additional 10 Mio messages then compact
+	for i := uint64(0); i < toStoreOnTop; i++ {
+		ms.StoreMsg(subj, nil, msg)
+	}
+	startFirstRun := time.Now()
+	purgedFirstRun, _ := ms.Compact(10_300_000)
+	elapsedFirstRun := time.Since(startFirstRun).Microseconds()
+	if purgedFirstRun != 299_999 {
```

Comment:
Again require_Equal is our standard for these.

Reply:
Understood & Fixed